buf should check if the file is actually protobuf file

During testing of a new (private) repository, the directory structure is:

```
/dir
- Makefile.proto
- a/
  - a.proto
- b/
  - b.proto
- c/
  - c.proto
```

The Makefile.proto file is not a protobuf file but actually a Makefile.  While that's clearly a problem,  buf throws a SEGV and this stack trace:

```
$ buf check lint
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x99371b]

goroutine 18 [running]:
github.com/jhump/protoreflect/desc/protoparse.(*basicCompositeNode).start(...)
github.com/jhump/protoreflect@v1.7.1-0.20200706205124-8d2d934ce565/desc/protoparse/ast.go:162
github.com/jhump/protoreflect/desc/protoparse.(*parseResult).createFileDescriptor(0xc00039c0f0, 0xc0002b6c30, 0xe, 0xc00038c180)
github.com/jhump/protoreflect@v1.7.1-0.20200706205124-8d2d934ce565/desc/protoparse/descriptor_protos.go:35 +0xd9b
github.com/jhump/protoreflect/desc/protoparse.createParseResult(0xc0002b6c30, 0xe, 0xc00038c180, 0xc00039c030, 0xd42f00)
github.com/jhump/protoreflect@v1.7.1-0.20200706205124-8d2d934ce565/desc/protoparse/parser.go:723 +0xba
github.com/jhump/protoreflect/desc/protoparse.parseProto(0xc0002b6c30, 0xe, 0x7f82285a80e0, 0xc00038e080, 0xc00039c030, 0xc0002b6c01, 0xe)
github.com/jhump/protoreflect@v1.7.1-0.20200706205124-8d2d934ce565/desc/protoparse/parser.go:702 +0x16e
github.com/jhump/protoreflect/desc/protoparse.parseProtoFile.func2(0x7f82285a80b8, 0xc00038e080, 0xc0002b6c30, 0xe, 0xc00039c030, 0x1, 0xc00008fb00)
github.com/jhump/protoreflect@v1.7.1-0.20200706205124-8d2d934ce565/desc/protoparse/parser.go:473 +0xba
github.com/jhump/protoreflect/desc/protoparse.parseProtoFile(0xc00039a000, 0xc0002b6c30, 0xe, 0x0, 0xc00039c030, 0x400101, 0xc00008fcc8, 0xc4d950)
github.com/jhump/protoreflect@v1.7.1-0.20200706205124-8d2d934ce565/desc/protoparse/parser.go:474 +0x958
github.com/jhump/protoreflect/desc/protoparse.parseProtoFiles(0xc00039a000, 0xc0002baa80, 0xd, 0xd, 0xc00039c030, 0x101, 0xc00008fcc8, 0x0)
github.com/jhump/protoreflect@v1.7.1-0.20200706205124-8d2d934ce565/desc/protoparse/parser.go:447 +0xb9
github.com/jhump/protoreflect/desc/protoparse.Parser.ParseFiles(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc00039a000, 0x1, 0xc00038e020, 0x0, ...)
github.com/jhump/protoreflect@v1.7.1-0.20200706205124-8d2d934ce565/desc/protoparse/parser.go:217 +0x211
github.com/bufbuild/buf/internal/buf/bufbuild.getBuildResult(0xd3b620, 0xc0000bb560, 0xc0001d53b0, 0xc0002baa80, 0xd, 0xd, 0x0, 0x0)
github.com/bufbuild/buf/internal/buf/bufbuild/builder.go:179 +0x176
github.com/bufbuild/buf/internal/buf/bufbuild.(*builder).getBuildResults.func1(0xc00030aa80, 0xd3b620, 0xc0000bb560, 0xc0001d53b0, 0xc0002baa80, 0xd, 0xd, 0x203000)
github.com/bufbuild/buf/internal/buf/bufbuild/builder.go:138 +0x73
created by github.com/bufbuild/buf/internal/buf/bufbuild.(*builder).getBuildResults
github.com/bufbuild/buf/internal/buf/bufbuild/builder.go:137 +0x211
```

This seems to indicate that there is a weakness in buf's parser when presented with non-proto files.  It should definitely handle that type of failure better.  Thanks!

--- a/go.mod
+++ b/go.mod
@@ -7,7 +7,7 @@ require (
 	github.com/gofrs/uuid v3.3.0+incompatible
 	github.com/golang/protobuf v1.4.2
 	github.com/google/go-cmp v0.5.0
-	github.com/jhump/protoreflect v1.7.1-0.20200706205124-8d2d934ce565
+	github.com/jhump/protoreflect v1.7.1-0.20200713144533-3216fce50c70
 	github.com/klauspost/compress v1.10.10
 	github.com/klauspost/pgzip v1.2.4
 	github.com/pkg/profile v1.5.0
@@ -16,7 +16,7 @@ require (
 	github.com/stretchr/testify v1.6.1
 	go.uber.org/multierr v1.5.0
 	go.uber.org/zap v1.15.0
-	google.golang.org/genproto v0.0.0-20200710124503-20a17af7bd0e // indirect
+	google.golang.org/genproto v0.0.0-20200711021454-869866162049 // indirect
 	google.golang.org/protobuf v1.25.0
 	gopkg.in/yaml.v3 v3.0.0-20200615113413-eeeca48fe776
 )

--- a/internal/buf/cmd/buf/buf.go
+++ b/internal/buf/cmd/buf/buf.go
@@ -22,7 +22,7 @@ import (
 )
 
 // Version is the version of buf.
-const Version = "0.19.1"
+const Version = "0.20.0-dev"
 
 // Main is the main.
 func Main(use string, options ...RootCommandOption) {

--- a/make/buf/base.mk
+++ b/make/buf/base.mk
@@ -1,6 +1,6 @@
 BUF_BIN ?= cmd/buf
 
-PROTOREFLECT_VERSION := 8d2d934ce5650f450ffe1f441313e26c75e1fdab
+PROTOREFLECT_VERSION := 3216fce50c7038421461bd10ee796f1dfcc82dcc
 GO_GET_PKGS := $(GO_GET_PKGS) github.com/jhump/protoreflect@$(PROTOREFLECT_VERSION)
 GO_BINS := $(GO_BINS) $(BUF_BIN) \
 	cmd/protoc-gen-buf-check-breaking \

